### ISSUE TITLE
[BUG] `cannot import name 'PYZ_TYPE_PKG'` when packing with pyinstaller

### ISSUE THREAD

**Issue by mevaere** [dashingsoft/pyarmor#1215]
main.py

```
from fibonacci import fibonacci

print('this is __pyarmor__', __pyarmor__)
print("Welcome to the test program")
nbr = input("Please enter an integer : ")
if None != (fib_list := fibonacci(nbr)):
    print("Sequence of Fibonacci : ")
    print(fib_list)
input("Press key to stop")
```

> pyinstaller main.py

```
139 INFO: PyInstaller: 5.10.1
139 INFO: Python: 3.10.11
145 INFO: Platform: Windows-10-10.0.22621-SP0
146 INFO: wrote D:\JetBrainsProjects\PyCharm\PyArmorV2\main.spec
147 INFO: UPX is not available.
148 INFO: Extending PYTHONPATH with paths
['D:\\JetBrainsProjects\\PyCharm\\PyArmorV2']
275 INFO: checking Analysis
275 INFO: Building Analysis because Analysis-00.toc is non existent
275 INFO: Initializing module dependency graph...
276 INFO: Caching module graph hooks...
283 INFO: Analyzing base_library.zip ...
932 INFO: Loading module hook 'hook-heapq.py' from 'D:\\JetBrainsProjects\\PyCharm\\PyArmorV2\\venv\\lib\\site-packages\\PyInstaller\\hooks'...
964 INFO: Loading module hook 'hook-encodings.py' from 'D:\\JetBrainsProjects\\PyCharm\\PyArmorV2\\venv\\lib\\site-packages\\PyInstaller\\hooks'...
1626 INFO: Loading module hook 'hook-pickle.py' from 'D:\\JetBrainsProjects\\PyCharm\\PyArmorV2\\venv\\lib\\site-packages\\PyInstaller\\hooks'...
2218 INFO: Caching module dependency graph...
2274 INFO: running Analysis Analysis-00.toc
2276 INFO: Adding Microsoft.Windows.Common-Controls to dependent assemblies of final executable
  required by C:\Users\remi\AppData\Local\Programs\Python\Python310\python.exe
2300 INFO: Analyzing D:\JetBrainsProjects\PyCharm\PyArmorV2\main.py
2301 INFO: Processing module hooks...
2308 INFO: Looking for ctypes DLLs
2309 INFO: Analyzing run-time hooks ...
2310 INFO: Including run-time hook 'D:\\JetBrainsProjects\\PyCharm\\PyArmorV2\\venv\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth_inspect.py'        
2312 INFO: Looking for dynamic libraries
172 INFO: Extra DLL search directories (AddDllDirectory): []
172 INFO: Extra DLL search directories (PATH): ['D:\\JetBrainsProjects\\PyCharm\\PyArmorV2\\venv\\Scripts', 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUD
A\\v12.1\\bin', 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v12.1\\libnvvp', 'C:\\Program Files (x86)\\Common Files\\Oracle\\Java\\javapath', 'C:\\Wi
ndows\\system32', 'C:\\Windows', 'C:\\Windows\\System32\\Wbem', 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\', 'C:\\Windows\\System32\\OpenSSH\\', 'C:\\Prog
ram Files\\Calibre2\\', 'C:\\Program Files (x86)\\Gpg4win\\..\\GnuPG\\bin', 'C:\\Users\\remi\\AppData\\Roaming\\Python\\Scripts', 'C:\\Program Files\\nodejs\\', 
'C:\\ProgramData\\chocolatey\\bin', 'C:\\Program Files\\Git\\cmd', 'C:\\Program Files (x86)\\NVIDIA Corporation\\PhysX\\Common', 'C:\\Program Files\\NVIDIA Corpo
ration\\NVIDIA NvDLISR', 'C:\\Program Files\\NVIDIA Corporation\\Nsight Compute 2023.1.1\\', 'C:\\Users\\remi\\AppData\\Local\\Programs\\Python\\Python311\\Scrip
ts\\', 'C:\\Users\\remi\\AppData\\Local\\Programs\\Python\\Python311\\', 'C:\\Users\\remi\\AppData\\Local\\Microsoft\\WindowsApps', 'C:\\Users\\remi\\AppData\\Lo
cal\\GitHubDesktop\\bin', 'C:\\Users\\remi\\AppData\\Local\\JetBrains\\Toolbox\\scripts', 'C:\\Program Files\\JetBrains\\PyCharm\\bin', 'C:\\Users\\remi\\AppData\\Roaming\\npm']
2685 INFO: Looking for eggs
2685 INFO: Using Python library C:\Users\remi\AppData\Local\Programs\Python\Python310\python310.dll
2686 INFO: Found binding redirects:
[]
2687 INFO: Warnings written to D:\JetBrainsProjects\PyCharm\PyArmorV2\build\main\warn-main.txt
2698 INFO: Graph cross-reference written to D:\JetBrainsProjects\PyCharm\PyArmorV2\build\main\xref-main.html
2710 INFO: checking PYZ
2710 INFO: Building PYZ because PYZ-00.toc is non existent
2710 INFO: Building PYZ (ZlibArchive) D:\JetBrainsProjects\PyCharm\PyArmorV2\build\main\PYZ-00.pyz
2844 INFO: Building PYZ (ZlibArchive) D:\JetBrainsProjects\PyCharm\PyArmorV2\build\main\PYZ-00.pyz completed successfully.
2847 INFO: checking PKG
2847 INFO: Building PKG because PKG-00.toc is non existent
2847 INFO: Building PKG (CArchive) main.pkg
2851 INFO: Building PKG (CArchive) main.pkg completed successfully.
2852 INFO: Bootloader D:\JetBrainsProjects\PyCharm\PyArmorV2\venv\lib\site-packages\PyInstaller\bootloader\Windows-64bit-intel\run.exe
2852 INFO: checking EXE
2852 INFO: Building EXE because EXE-00.toc is non existent
2852 INFO: Building EXE from EXE-00.toc
2852 INFO: Copying bootloader EXE to D:\JetBrainsProjects\PyCharm\PyArmorV2\build\main\main.exe.notanexecutable
2854 INFO: Copying icon to EXE
2854 INFO: Copying icons from ['D:\\JetBrainsProjects\\PyCharm\\PyArmorV2\\venv\\lib\\site-packages\\PyInstaller\\bootloader\\images\\icon-console.ico']
2863 INFO: Writing RT_GROUP_ICON 0 resource with 104 bytes
2863 INFO: Writing RT_ICON 1 resource with 3752 bytes
2863 INFO: Writing RT_ICON 2 resource with 2216 bytes
2863 INFO: Writing RT_ICON 3 resource with 1384 bytes
2863 INFO: Writing RT_ICON 4 resource with 37019 bytes
2863 INFO: Writing RT_ICON 5 resource with 9640 bytes
2863 INFO: Writing RT_ICON 6 resource with 4264 bytes
2863 INFO: Writing RT_ICON 7 resource with 1128 bytes
2864 INFO: Copying 0 resources to EXE
2864 INFO: Embedding manifest in EXE
2864 INFO: Updating manifest in D:\JetBrainsProjects\PyCharm\PyArmorV2\build\main\main.exe.notanexecutable
2871 INFO: Updating resource type 24 name 1 language 0
2872 INFO: Appending PKG archive to EXE
2880 INFO: Fixing EXE headers
3086 INFO: Building EXE from EXE-00.toc completed successfully.
3088 INFO: checking COLLECT
3088 INFO: Building COLLECT because COLLECT-00.toc is non existent
3088 INFO: Building COLLECT COLLECT-00.toc
3213 INFO: Building COLLECT COLLECT-00.toc completed successfully.
```

> pyarmor gen --pack dist/main/main.exe main.py

```
INFO     search inputs ...
INFO     find script main.py
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'windows.amd64'}
INFO     write dist\pyarmor_runtime_005107\pyarmor_runtime.pyd
INFO     patch runtime file
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "main"
INFO     obfuscating file main.py
INFO     write dist\main.py
INFO     obfuscate scripts OK
ERROR    cannot import name 'PYZ_TYPE_PKG' from 'PyInstaller.loader.pyimod01_archive' (D:\JetBrainsProjects\PyCharm\PyArmorV2\venv\lib\site-packages\PyInstaller\loader\pyimod01_archive.py)
```

Seems to work with manual patch.

Kind regards

```
INFO     Python 3.10.11
INFO     Pyarmor 8.1.9 (pro)
INFO     Platform windows.x86_64
```

**Comment by jondy** [comment]
> Seems to work with manual patch.
Does it mean change `PYZ_TYPE_PKG` to `PYZ_ITEM_PKG`?

**Comment by jondy** [comment]
Could you verify this patch works or not?
056facd058460240040d2aca854441692af2d8be

I haven't PyInstaller 5.10 installed, only PyInstaller 5.6

**Comment by mevaere** [comment]
It doesn't work

```
INFO     Python 3.10.11
INFO     Pyarmor 8.1.9 (pro), 005107, MElectronSoft
INFO     Platform windows.x86_64
INFO     search inputs ...
INFO     find script main.py
INFO     find script fibonacci.py
INFO     find 2 top resources
INFO     start to generate runtime files
INFO     target platforms {'windows.amd64'}
INFO     write dist\pyarmor_runtime_005107\pyarmor_runtime.pyd
INFO     patch runtime file
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "main"
INFO     obfuscating file main.py
INFO     write dist\main.py
INFO     process resource "fibonacci"
INFO     obfuscating file fibonacci.py
INFO     write dist\fibonacci.py
INFO     obfuscate scripts OK
INFO     repack bundle "dist/main/main.exe"
INFO     obfuscated scripts at "dist"
INFO     entry script name is "main.py"
INFO     runtime package is "pyarmor_runtime_005107"
INFO     extracted bundle files to ".pyarmor\pack\main_extracted"
ERROR    'CArchiveReader' object has no attribute 'lib'
```

**Comment by jondy** [comment]
Got it, thanks. It need more code to support PyInstaller 5.10

**Comment by jondy** [comment]
Fixed in 8.2.0
